server/container/util/src/main/java/org/apache/james/util/scanner/SpamAssassinInvoker.java

```diff
@@ -78,96 +68,60 @@ public SpamAssassinInvoker(String spamdHost, int spamdPort) {
      * @throws MessagingException
      *             if an error on scanning is detected
      */
-    public boolean scanMail(MimeMessage message) throws MessagingException {
-        Socket socket = null;
-        OutputStream out = null;
-        BufferedReader in = null;
-
-        try {
-            socket = new Socket(spamdHost, spamdPort);
+    public SpamAssassinResult scanMail(MimeMessage message) throws MessagingException {
+        try (Socket socket = new Socket(spamdHost, spamdPort);
+                OutputStream out = socket.getOutputStream();
+                BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
 
-            out = socket.getOutputStream();
-            in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
             out.write("CHECK SPAMC/1.2\r\n\r\n".getBytes());
```

Comment:
using a PrintWriter would avoid the ugly getBytes (and also forgetting the charset)